include/irr/asset/filters/kernels/CChannelIndependentImageFilterKernel.h

```diff
@@ -151,30 +198,58 @@ class CChannelIndependentImageFilterKernel<KernelR,KernelG,KernelB,void> : publi
 			return 0.f;
 		}
 
-		_IRR_STATIC_INLINE_CONSTEXPR bool has_derivative = KernelR::has_derivative&&KernelG::has_derivative&&KernelB::has_derivative;
-		inline std::enable_if_t<has_derivative,float> d_weight(float x, int32_t channel) const
+		_IRR_STATIC_INLINE_CONSTEXPR bool has_derivative = KernelR::has_derivative||KernelG::has_derivative||KernelB::has_derivative;
+		inline float d_weight(float x, int32_t channel) const
 		{
 			switch (channel)
 			{
 				case 0:
-					return KernelR::d_weight(x,0);
+					if constexpr (KernelR::has_derivative)
+					{
+						return KernelR::d_weight(x, 0);
+					}
+					_IRR_DEBUG_BREAK_IF(!KernelR::has_derivative);
+					break;
 				case 1:
-					return KernelG::d_weight(x,1);
+					if constexpr (KernelG::has_derivative)
+					{
+						return KernelG::d_weight(x, 1);
+					}
+					_IRR_DEBUG_BREAK_IF(!KernelG::has_derivative);
+					break;
 				case 2:
-					return KernelB::d_weight(x,2);
+					if constexpr (KernelB::has_derivative)
+					{
+						return KernelB::d_weight(x, 2);
+					}
+					_IRR_DEBUG_BREAK_IF(!KernelB::has_derivative);
+					break;
 				default:
 					break;
 			}
 			return 0.f;
 		}
+
+		IRR_DECLARE_DEFINE_CIMAGEFILTER_KERNEL_PASS_THROUGHS(base_t)
 };
 
 template<class KernelR, class KernelG, class KernelB, class KernelA>
-class CChannelIndependentImageFilterKernel : public CFloatingPointSeparableImageFilterKernelBase<CChannelIndependentImageFilterKernel<KernelR,KernelG,KernelB,KernelA>>, private KernelR,KernelG,KernelB,KernelA
+class CChannelIndependentImageFilterKernel : 
+	public CFloatingPointSeparableImageFilterKernelBase<CChannelIndependentImageFilterKernel<KernelR,KernelG,KernelB,KernelA>>, public impl::CChannelIndependentImageFilterKernelBase, private KernelR,KernelG,KernelB,KernelA
 {
+		using base_t = CFloatingPointSeparableImageFilterKernelBase<CChannelIndependentImageFilterKernel<KernelR, KernelG, KernelB, KernelA>>;
+
 	public:
+		using value_type = typename base_t::value_type;
+
 		_IRR_STATIC_INLINE_CONSTEXPR auto MaxChannels = 4;
 
+		CChannelIndependentImageFilterKernel(float _negative_support, float _positive_support, KernelR&& kernel_r, KernelG&& kernel_g, KernelB&& kernel_b, KernelA&& kernel_a) :
```

Comment:
hmmm the constructors should just take the r-value kernels and work out the negative and positive support (in all 3 dimensions) as the maximum of all kernel's supports.